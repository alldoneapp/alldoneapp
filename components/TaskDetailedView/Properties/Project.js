import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import ProjectPicker from './ProjectPicker'
import styles, { colors } from '../../styles/global'
import Icon from '../../Icon'
import { translate } from '../../../i18n/TranslationService'

export default function Project({ project, item, disabled }) {
    const isGuide = !!project.parentTemplateId
    return (
        <View style={localStyles.container}>
            <View style={localStyles.projectBallContainer}>
                <Icon name="circle" size={24} color={colors.Text03} />
            </View>
            <Text style={[styles.subtitle2, { color: colors.Text03 }]}>{translate('Project')}</Text>
            <View style={{ marginLeft: 'auto' }}>
                <ProjectPicker project={project} item={item} disabled={disabled || isGuide} />
            </View>
        </View>
    )
}

const localStyles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'row',
        maxHeight: 56,
        minHeight: 56,
        height: 56,
        paddingLeft: 8,
        paddingVertical: 8,
        alignItems: 'center',
    },
    projectBallContainer: {
        marginRight: 8,
        width: 24,
        height: 24,
        alignItems: 'center',
        justifyContent: 'center',
    },
    projectMarker: {
        width: 16,
        height: 16,
        borderRadius: 100,
    },
})
