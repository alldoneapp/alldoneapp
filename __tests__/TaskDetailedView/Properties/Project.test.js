/**
 * @jest-environment jsdom
 */

import React from 'react'
import Project from '../../../components/TaskDetailedView/Properties/Project'

import renderer from 'react-test-renderer'

jest.mock('firebase', () => ({ firestore: {} }));
jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn().mockImplementation(fnc => {
        return fnc({
            smallScreenNavigation: {}
        })
    }),
}));

describe('Project component', () => {
    describe('Project snapshot test', () => {
        it('should render correctly', () => {
            const tree = renderer.create(<Project project={{ name: '', color: '' }} />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
