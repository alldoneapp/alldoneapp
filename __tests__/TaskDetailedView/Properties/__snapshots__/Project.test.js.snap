// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project component Project snapshot test should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "maxHeight": 56,
      "paddingLeft": 11,
      "paddingVertical": 8,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 24,
        "justifyContent": "center",
        "marginRight": 8,
        "width": 24,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#8A94A6",
            "fontFamily": "alldone",
            "fontSize": 22,
            "opacity": 1,
          }
        }
      >
        î¤¹
      </Text>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#04142F",
          "fontFamily": "Roboto-Medium",
          "fontSize": 14,
          "letterSpacing": 0.16,
          "lineHeight": 22,
        },
        Object {
          "color": "#8A94A6",
        },
      ]
    }
  >
    Project
  </Text>
  <View
    style={
      Object {
        "marginLeft": "auto",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "transparent",
          "borderColor": "#C6CDD2",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 1,
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "height": 40,
          "justifyContent": "center",
          "maxHeight": 40,
          "maxWidth": 240,
          "minHeight": 40,
          "opacity": 1,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingVertical": 8,
        }
      }
    >
      
    </View>
  </View>
</View>
`;
